Shop/Shop/Controllers/ProductController.cs

```diff
@@ -27,20 +29,33 @@ public List<Product> GetAll()
         }
 
         [HttpDelete("{id}")]
-        public IActionResult Delete(int id)
+        public Product Delete(int id)
         {
+            if (IsCorrectId(id) is null)
```

Comment:
code is hard to read. there is should be one check, like if(IsCorrectId(id)) { ... } without additional check on null